test/unit/slurm/test_slurm_config_generator.py

```diff
@@ -9,13 +9,19 @@
 # OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
 # limitations under the License.
 
+import inspect
 import os
 
 import pytest
 import slurm
 from assertpy import assert_that
-from slurm.pcluster_slurm_config_generator import generate_slurm_config_files
+from pathlib import Path
+from pcluster_slurm_config_generator import generate_slurm_config_files
 
+def _get_template_folder() -> str:
+    module_file = inspect.getsourcefile(generate_slurm_config_files)
+    template_path = Path(module_file).parent / "templates"
+    return str(template_path)
```

Comment:
Is this a solid implementation to retrieve the templates folder? I'm asking because I honestly don't know.

Still, I guess it should be equivalent to the old `os.path.dirname(slurm.__file__) + "/templates"`

Reply:
It is equivalent "by design" :)

It is also a solid implementation and it is more general since starts from the path of a module containing a given function. As of now since we only have one use I embedded the function name, and so the module.
However if other cookbooks need a similar function it could be moved in common and passing the function as parameter it would work also for other paths/modules/cookbooks if they are in the python_path of the code that is invoking the function.